File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        i = 0
        for num in nums:
            diff = target - num
            if diff in hashMap:
                return [i, hashMap[diff]]
            hashMap[num] = i
            i += 1


        

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashMap;
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            int diff = target - nums[i];
            if (hashMap.find(diff) != hashMap.end()){
                return {hashMap[diff], i};
            }
            hashMap.insert({nums[i], i});
        }
        return {};
    }
};